Patch: Update the review type

estib-vega opened the pull request:
Update the review properties of patch to match the new user information returned by the API

Depends on https://github.com/gitbutlerapp/web/pull/460

estib-vega committed:
Patch: Update the review type

Update the review properties of patch to match the new user information returned by the API

estib-vega committed:
Adapt the patch reviewers group

Add a new function for getting all the viewers' avatar information and use it in the PatchReviewersGroup component.

Caleb-T-Owens commented on packages/shared/src/lib/branches/types.ts:
```diff
@@ -135,23 +135,49 @@ export function apiToPatchStatistics(api: ApiPatchStatistics): PatchStatistics {
+	name: string | null;
+};
+
+export type PatchReviewUser = {
+	id: number;
+	avatarUrl: string | undefined;
+	email: string | undefined;
+	login: string | undefined;
+	name: string | undefined;
+};
```
Rather than calling this a PatchReviewUser, we should properly match the SimpleUser > User > PrivateUser higherarchy from the backend

estib-vega replied:
Done. I created a central type under the `users` directory

estib-vega committed:
User types: Centralize the user definitions

Reuse the same type and formatter functions for the user entities in order to match the backend entities and have a single source of truth.
